howso/train.amlg

```diff
@@ -780,79 +763,123 @@
+										feature_values feature_values
+										features features
+									))
+							))
+						)
+						;do not train on this case if it is null or all case values are null or it's within provided thresholds
+						;if one of the ablation methods returns false, then the case should be ablated.
+						(and
+							(call !CaseOutsideThresholds)
+							(call !ShouldNewCaseBeAblated (assoc
+								features features
+								feature_values feature_values
+								;always use the most current trained count to determine whether cases should be ablated
+								num_cases (+ ablation_trained_instance_count case_index)
+							))
+						)
 					))
+					(indices input_cases)
 				)
-				;do not ablate, create the case and store case id
-				(let
-					(assoc
-						new_case_id
-							(call !CreateCase (assoc
-								features train_features
+		)
+
+		;ensure ablated indices are based off actual training index and not restarted at 0 every time this method is called
+		(accum (assoc
+			ablated_indices_list
+				(map
+					(lambda (+ session_training_index (current_value)))
+					(remove (indices input_cases) indices_to_train)
+				)
+			ablation_trained_instance_count (size indices_to_train)
+		))
+
+		;create the cases
+		(declare (assoc
+			output_cases
+				(map
+					(lambda (let
+						(assoc feature_values (get input_cases (current_value 1)) )
+						(if encode_features_on_train
+							(assign (assoc
 								feature_values
-									;if auto ablate is enabled, populate the weight feature for this case
-									(if !autoAblationEnabled
-										(append feature_values (list 1))
-										feature_values
-									)
-								session (get_value session)
-								session_training_index (+ session_training_index case_index)
+									(call !ConvertFromInput (assoc
+										feature_values feature_values
+										features features
+									))
 							))
-					)
+						)
 
-					;if auto analyzing, need to accumulate data mass after every created case
-					(if run_autoanalyze_check
-						(accum_to_entities (assoc
-							!dataMassChangeSinceLastAnalyze
-								(if accumulate_weight_feature
-									(or (get feature_values weight_feature_index) 1)
-									1
+						;create the case and output case id
+						(call !CreateCase (assoc
+							features train_features
+							feature_values
+								;if auto ablate is enabled, populate the weight feature for this case
+								(if !autoAblationEnabled
+									(append feature_values (list 1))
+									feature_values
 								)
+							session (get_value session)
+							session_training_index (+ session_training_index (current_value 1))
 						))
-					)
+					))
+					indices_to_train
+				)
+		))
 
-					(accum (assoc ablation_trained_instance_count 1))
+		;if auto analyzing, need to accumulate data masses for all cases
+		(if run_autoanalyze_check
+			(let
+				(assoc
+					mass_to_accumulate
+						(if (and accumulate_weight_feature (size indices_to_train))
```

Comment:
this should probably be `(if (and weight_feature_index (size indices_to_train))`.

I don't see why it would be `accumulate_weight_feature`. Looks like it may have been incorrect before too if this was the case. Probably also worth verifying that `weight_feature_index` is available here as I think I saw that was moved.

Reply:
no, all the logic is based off accumulate_weight_feature, that is the boolean to do the accumulation. weight_feature_index could be 0.